### ISSUE TITLE
Unable to authenticate using Google Apps Pro

### ISSUE THREAD

**Issue by faichelbaum** [xwikisas/application-googleapps#5]
Hi

we upgraded this morning to 10.7. Since then, the former extension Google Apps (2.1.8) is not working. 
We removed this app and installed the new Pro version (2.3.3) but we struggle to login.

We end up with this error:

> Failed to execute the [velocity] macro. Cause: [startup failed: script15362362231101439713831.groovy: 346: You cannot create an instance from the abstract class 'com.google.api.services.plus.Plus$Builder'. @ line 346, column 16. return new Plus.Builder( ^ script15362362231101439713831.groovy: 355: You cannot create an instance from the abstract class 'com.google.api.services.plus.Plus$Builder'. @ line 355, column 16. def plus = new Plus.Builder( ^ script15362362231101439713831.groovy: 459: You cannot create an instance from the abstract class 'com.google.api.services.drive.Drive$Builder'. @ line 459, column 16. return new Drive.Builder( ^ 3 errors ]. Click on this message for details.

Displaying the complete trace provide the following:

> org.xwiki.rendering.macro.MacroExecutionException: Failed to evaluate Velocity Macro for content [#set($gagroovy = $xwiki.parseGroovyFromPage("GoogleApps.Groovy"))
> #set($ok = $gagroovy.init($xwiki, $context, $doc, $services))
> #set($result = $gagroovy.updateUser())
> #if($result==-1)
> #set($user = $gagroovy.getGoogleUser())
> #set($email = $user.emails[0].value)
> ${services.localization.render("googleapps.login.domainerror1")}
> 
> ${services.localization.render("googleapps.login.domainerror2")} $!{email}.
> ${services.localization.render("googleapps.login.domainerror3")}
> 
> #elseif($result)
> ${services.localization.render("googleapps.login.redirectmessage")}
> 
>   #if($request.xredirect && $request.xredirect!="")
>     $response.sendRedirect($request.xredirect)
>   #else
>     $response.sendRedirect($xwiki.getURL("Main.WebHome"))
>   #end
> #else
> ${services.localization.render("googleapps.login.message")}
> 
>   ${services.localization.render("googleapps.login.error")}
> #end]
> 	at org.xwiki.rendering.internal.macro.velocity.VelocityMacro.evaluateString(VelocityMacro.java:139)
> 	at org.xwiki.rendering.internal.macro.velocity.VelocityMacro.evaluateString(VelocityMacro.java:52)
> 	at org.xwiki.rendering.macro.script.AbstractScriptMacro.evaluateBlock(AbstractScriptMacro.java:286)
> 	at org.xwiki.rendering.macro.script.AbstractScriptMacro.execute(AbstractScriptMacro.java:182)
> 	at org.xwiki.rendering.macro.script.AbstractScriptMacro.execute(AbstractScriptMacro.java:58)
> 	at org.xwiki.rendering.internal.transformation.macro.MacroTransformation.transform(MacroTransformation.java:297)
> 	at org.xwiki.rendering.internal.transformation.DefaultRenderingContext.transformInContext(DefaultRenderingContext.java:183)
> 	at org.xwiki.rendering.internal.transformation.DefaultTransformationManager.performTransformations(DefaultTransformationManager.java:101)
> 	at org.xwiki.display.internal.DocumentContentDisplayer.display(DocumentContentDisplayer.java:263)
> 	at org.xwiki.display.internal.DocumentContentDisplayer.display(DocumentContentDisplayer.java:133)
> 	at org.xwiki.display.internal.DocumentContentDisplayer.display(DocumentContentDisplayer.java:58)
> 	at org.xwiki.display.internal.DefaultDocumentDisplayer.display(DefaultDocumentDisplayer.java:96)
> 	at org.xwiki.display.internal.DefaultDocumentDisplayer.display(DefaultDocumentDisplayer.java:39)
> 	at org.xwiki.sheet.internal.SheetDocumentDisplayer.display(SheetDocumentDisplayer.java:123)
> 	at org.xwiki.sheet.internal.SheetDocumentDisplayer.display(SheetDocumentDisplayer.java:52)
> 	at org.xwiki.display.internal.ConfiguredDocumentDisplayer.display(ConfiguredDocumentDisplayer.java:68)
> 	at org.xwiki.display.internal.ConfiguredDocumentDisplayer.display(ConfiguredDocumentDisplayer.java:42)
> 	at com.xpn.xwiki.doc.XWikiDocument.display(XWikiDocument.java:1179)
> 	at com.xpn.xwiki.doc.XWikiDocument.getRenderedContent(XWikiDocument.java:1220)
> 	at com.xpn.xwiki.doc.XWikiDocument.getRenderedContent(XWikiDocument.java:1197)
> 	at com.xpn.xwiki.doc.XWikiDocument.getRenderedContent(XWikiDocument.java:1228)
> 	at com.xpn.xwiki.api.Document.getRenderedContent(Document.java:730)
> 	at sun.reflect.GeneratedMethodAccessor627.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:395)
> 	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:384)
> 	at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:173)
> 	at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:280)
> 	at org.apache.velocity.runtime.parser.node.ASTReference.value(ASTReference.java:567)
> 	at org.apache.velocity.runtime.parser.node.ASTExpression.value(ASTExpression.java:71)
> 	at org.apache.velocity.runtime.parser.node.ASTSetDirective.render(ASTSetDirective.java:142)
> 	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
> 	at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:342)
> 	at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:106)
> 	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
> 	at org.xwiki.velocity.introspection.TryCatchDirective.render(TryCatchDirective.java:87)
> 	at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:207)
> 	at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:342)
> 	at org.xwiki.velocity.internal.DefaultVelocityEngine.evaluateInternal(DefaultVelocityEngine.java:259)
> 	at org.xwiki.velocity.internal.DefaultVelocityEngine.evaluate(DefaultVelocityEngine.java:222)
> 	at com.xpn.xwiki.render.DefaultVelocityManager.evaluate(DefaultVelocityManager.java:357)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.evaluateContent(InternalTemplateManager.java:825)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:701)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.lambda$renderFromSkin$0(InternalTemplateManager.java:676)
> 	at com.xpn.xwiki.internal.security.authorization.DefaultAuthorExecutor.call(DefaultAuthorExecutor.java:85)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.renderFromSkin(InternalTemplateManager.java:675)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.renderFromSkin(InternalTemplateManager.java:654)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:640)
> 	at com.xpn.xwiki.internal.template.DefaultTemplateManager.render(DefaultTemplateManager.java:78)
> 	at com.xpn.xwiki.XWiki.evaluateTemplate(XWiki.java:2269)
> 	at com.xpn.xwiki.XWiki.parseTemplate(XWiki.java:2247)
> 	at com.xpn.xwiki.api.XWiki.parseTemplate(XWiki.java:992)
> 	at sun.reflect.GeneratedMethodAccessor296.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:395)
> 	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:384)
> 	at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:173)
> 	at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:280)
> 	at org.apache.velocity.runtime.parser.node.ASTReference.render(ASTReference.java:369)
> 	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
> 	at org.apache.velocity.runtime.directive.VelocimacroProxy.render(VelocimacroProxy.java:216)
> 	at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:311)
> 	at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:230)
> 	at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:207)
> 	at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:342)
> 	at org.xwiki.velocity.internal.DefaultVelocityEngine.evaluateInternal(DefaultVelocityEngine.java:259)
> 	at org.xwiki.velocity.internal.DefaultVelocityEngine.evaluate(DefaultVelocityEngine.java:222)
> 	at com.xpn.xwiki.render.DefaultVelocityManager.evaluate(DefaultVelocityManager.java:357)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.evaluateContent(InternalTemplateManager.java:825)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:701)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.lambda$renderFromSkin$0(InternalTemplateManager.java:676)
> 	at com.xpn.xwiki.internal.security.authorization.DefaultAuthorExecutor.call(DefaultAuthorExecutor.java:85)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.renderFromSkin(InternalTemplateManager.java:675)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.renderFromSkin(InternalTemplateManager.java:654)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:640)
> 	at com.xpn.xwiki.internal.template.DefaultTemplateManager.render(DefaultTemplateManager.java:78)
> 	at com.xpn.xwiki.XWiki.evaluateTemplate(XWiki.java:2269)
> 	at com.xpn.xwiki.XWiki.parseTemplate(XWiki.java:2247)
> 	at com.xpn.xwiki.api.XWiki.parseTemplate(XWiki.java:992)
> 	at sun.reflect.GeneratedMethodAccessor296.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:395)
> 	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:384)
> 	at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:173)
> 	at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:280)
> 	at org.apache.velocity.runtime.parser.node.ASTReference.render(ASTReference.java:369)
> 	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
> 	at org.apache.velocity.runtime.directive.VelocimacroProxy.render(VelocimacroProxy.java:216)
> 	at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:311)
> 	at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:230)
> 	at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:207)
> 	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
> 	at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:87)
> 	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
> 	at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:342)
> 	at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:106)
> 	at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:342)
> 	at org.xwiki.velocity.internal.DefaultVelocityEngine.evaluateInternal(DefaultVelocityEngine.java:259)
> 	at org.xwiki.velocity.internal.DefaultVelocityEngine.evaluate(DefaultVelocityEngine.java:222)
> 	at com.xpn.xwiki.render.DefaultVelocityManager.evaluate(DefaultVelocityManager.java:357)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.evaluateContent(InternalTemplateManager.java:825)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:701)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.lambda$renderFromSkin$0(InternalTemplateManager.java:676)
> 	at com.xpn.xwiki.internal.security.authorization.DefaultAuthorExecutor.call(DefaultAuthorExecutor.java:85)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.renderFromSkin(InternalTemplateManager.java:675)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.renderFromSkin(InternalTemplateManager.java:654)
> 	at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:640)
> 	at com.xpn.xwiki.internal.template.DefaultTemplateManager.render(DefaultTemplateManager.java:78)
> 	at com.xpn.xwiki.XWiki.evaluateTemplate(XWiki.java:2269)
> 	at com.xpn.xwiki.web.Utils.parseTemplate(Utils.java:180)
> 	at com.xpn.xwiki.web.XWikiAction.execute(XWikiAction.java:475)
> 	at com.xpn.xwiki.web.XWikiAction.execute(XWikiAction.java:209)
> 	at org.apache.struts.action.RequestProcessor.processActionPerform(RequestProcessor.java:425)
> 	at org.apache.struts.action.RequestProcessor.process(RequestProcessor.java:228)
> 	at org.apache.struts.action.ActionServlet.process(ActionServlet.java:1913)
> 	at org.apache.struts.action.ActionServlet.doGet(ActionServlet.java:449)
> 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:635)
> 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:742)
> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
> 	at com.xpn.xwiki.web.ActionFilter.doFilter(ActionFilter.java:112)
> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
> 	at org.xwiki.wysiwyg.filter.ConversionFilter.doFilter(ConversionFilter.java:127)
> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
> 	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
> 	at org.xwiki.container.servlet.filters.internal.SetHTTPHeaderFilter.doFilter(SetHTTPHeaderFilter.java:63)
> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
> 	at org.xwiki.container.servlet.filters.internal.SavedRequestRestorerFilter.doFilter(SavedRequestRestorerFilter.java:208)
> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
> 	at org.xwiki.container.servlet.filters.internal.SetCharacterEncodingFilter.doFilter(SetCharacterEncodingFilter.java:111)
> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
> 	at org.xwiki.resource.servlet.RoutingFilter.doFilter(RoutingFilter.java:134)
> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
> 	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198)
> 	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
> 	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)
> 	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
> 	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
> 	at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:685)
> 	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:650)
> 	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
> 	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
> 	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:800)
> 	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
> 	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:806)
> 	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1498)
> 	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
> 	at java.lang.Thread.run(Thread.java:748)
> Caused by: org.xwiki.velocity.XWikiVelocityException: Failed to evaluate content with id [xwiki:GoogleApps.Login]
> 	at org.xwiki.velocity.internal.DefaultVelocityEngine.evaluate(DefaultVelocityEngine.java:227)
> 	at com.xpn.xwiki.render.DefaultVelocityManager.evaluate(DefaultVelocityManager.java:357)
> 	at org.xwiki.rendering.internal.macro.velocity.VelocityMacro.evaluateString(VelocityMacro.java:131)
> 	... 161 more
> Caused by: org.apache.velocity.exception.MethodInvocationException: Invocation of method 'parseGroovyFromPage' in  class com.xpn.xwiki.api.XWiki threw exception com.xpn.xwiki.XWikiException: Error number 14001 in 14: Failed compiling groovy script at 40:xwiki:GoogleApps.Login[line 1, column 25]
> 	at org.apache.velocity.runtime.parser.node.ASTMethod.handleInvocationException(ASTMethod.java:243)
> 	at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:187)
> 	at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:280)
> 	at org.apache.velocity.runtime.parser.node.ASTReference.value(ASTReference.java:567)
> 	at org.apache.velocity.runtime.parser.node.ASTExpression.value(ASTExpression.java:71)
> 	at org.apache.velocity.runtime.parser.node.ASTSetDirective.render(ASTSetDirective.java:142)
> 	at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:342)
> 	at org.xwiki.velocity.internal.DefaultVelocityEngine.evaluateInternal(DefaultVelocityEngine.java:259)
> 	at org.xwiki.velocity.internal.DefaultVelocityEngine.evaluate(DefaultVelocityEngine.java:222)
> 	... 163 more
> Caused by: com.xpn.xwiki.XWikiException: Error number 14001 in 14: Failed compiling groovy script
> 	at com.xpn.xwiki.internal.render.groovy.ParseGroovyFromString.parseGroovyFromString(ParseGroovyFromString.java:120)
> 	at com.xpn.xwiki.XWiki.parseGroovyFromString(XWiki.java:6383)
> 	at com.xpn.xwiki.api.XWiki.parseGroovyFromPage(XWiki.java:2671)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:395)
> 	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:384)
> 	at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:173)
> 	... 170 more
> Caused by: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
> script15362362231101439713831.groovy: 346: You cannot create an instance from the abstract class 'com.google.api.services.plus.Plus$Builder'.
>  @ line 346, column 16.
>            return new Plus.Builder(
>                   ^
> 
> script15362362231101439713831.groovy: 355: You cannot create an instance from the abstract class 'com.google.api.services.plus.Plus$Builder'.
>  @ line 355, column 16.
>        def plus = new Plus.Builder(
>                   ^
> 
> script15362362231101439713831.groovy: 459: You cannot create an instance from the abstract class 'com.google.api.services.drive.Drive$Builder'.
>  @ line 459, column 16.
>            return new Drive.Builder(
>                   ^
> 
> 3 errors
> 
> 	at org.codehaus.groovy.control.ErrorCollector.failIfErrors(ErrorCollector.java:311)
> 	at org.codehaus.groovy.control.CompilationUnit.applyToSourceUnits(CompilationUnit.java:983)
> 	at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:633)
> 	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:582)
> 	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:352)
> 	at groovy.lang.GroovyClassLoader.access$300(GroovyClassLoader.java:85)
> 	at groovy.lang.GroovyClassLoader$5.provide(GroovyClassLoader.java:321)
> 	at groovy.lang.GroovyClassLoader$5.provide(GroovyClassLoader.java:318)
> 	at org.codehaus.groovy.runtime.memoize.ConcurrentCommonCache.getAndPut(ConcurrentCommonCache.java:147)
> 	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:316)
> 	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:305)
> 	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:247)
> 	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:257)
> 	at com.xpn.xwiki.internal.render.groovy.ParseGroovyFromString.parseGroovyFromString(ParseGroovyFromString.java:111)
> 	... 179 more

Any idea ?

**Comment by acotiuga** [comment]
Hi @faichelbaum! We are investigating the issue and we'll update this page as soon as we have news. Thanks for reporting it.

**Comment by mflorea** [comment]
This seems to be [a Groovy problem](https://issues.apache.org/jira/browse/GROOVY-8632) that has been introduced in XWiki 10.5RC1 with [the upgrade to Groovy 2.5.0](https://github.com/xwiki/xwiki-commons/commit/17fc3b88be8472cb2824e7a0855fe23439571bd3). The problem will be fixed in XWiki 10.8 due to [the upgrade to Groovy 2.5.2](https://github.com/xwiki/xwiki-commons/commit/d229b9c3d1b7c52569f023d447842f9d0503a28c).

Groovy is a core dependency of XWiki so upgrading it at runtime through the Extension Manager is not possible. You need to upgrade the Groovy jars from the file system (``WEB-INF/lib``) and restart XWiki.

To summarize, this is a **platform** issue affecting XWiki versions 10.5 to 10.7. The fix is to upgrade Groovy to version 2.5.1+. I don't see what we could do on Google Apps Integration (Pro) extension to fix this issue.
